beacon_node/execution_layer/src/engine_api/http.rs

```diff
@@ -48,6 +49,480 @@ pub const BUILDER_GET_PAYLOAD_HEADER_TIMEOUT: Duration = Duration::from_secs(2);
+        u64::from_str_radix(strip_prefix(hex)?, 16)
+            .map_err(|e| format!("Failed to parse hex as u64: {:?}", e))
+    }
+
+    /// Parses a `0x`-prefixed, big-endian hex string as bytes.
+    ///
+    /// E.g., `0x0102 == vec![1, 2]`
+    fn hex_to_bytes(hex: &str) -> Result<Vec<u8>, String> {
+        hex::decode(strip_prefix(hex)?)
+            .map_err(|e| format!("Failed to parse hex as bytes: {:?}", e))
+    }
+
+    /// Removes the `0x` prefix from some bytes. Returns an error if the prefix is not present.
+    fn strip_prefix(hex: &str) -> Result<&str, String> {
+        if let Some(stripped) = hex.strip_prefix("0x") {
+            Ok(stripped)
+        } else {
+            Err("Hex string did not start with `0x`".to_string())
+        }
+    }
+
+    impl HttpJsonRpc<EngineApi> {
+        /// Get the eth1 chain id of the given endpoint.
+        pub async fn get_chain_id(&self, timeout: Duration) -> Result<Eth1Id, String> {
+            let chain_id: String = self
+                .rpc_request("eth_chainId", json!([]), timeout)
+                .await
+                .map_err(|e| format!("eth_chainId call failed {:?}", e))?;
+            hex_to_u64_be(chain_id.as_str()).map(|id| id.into())
+        }
+
+        /// Returns the current block number.
+        pub async fn get_block_number(&self, timeout: Duration) -> Result<u64, String> {
+            let response: String = self
+                .rpc_request("eth_blockNumber", json!([]), timeout)
+                .await
+                .map_err(|e| format!("eth_blockNumber call failed {:?}", e))?;
+            hex_to_u64_be(response.as_str())
+                .map_err(|e| format!("Failed to get block number: {}", e))
+        }
+
+        /// Gets a block hash by block number.
+        pub async fn get_block(
+            &self,
+            query: BlockQuery,
+            timeout: Duration,
+        ) -> Result<Block, String> {
+            let query_param = match query {
+                BlockQuery::Number(block_number) => format!("0x{:x}", block_number),
+                BlockQuery::Latest => "latest".to_string(),
+            };
+            let params = json!([
+                query_param,
+                false // do not return full tx objects.
+            ]);
+
+            let response: Value = self
+                .rpc_request("eth_getBlockByNumber", params, timeout)
+                .await
+                .map_err(|e| format!("eth_getBlockByNumber call failed {:?}", e))?;
+
+            let hash: Vec<u8> = hex_to_bytes(
+                response
+                    .get("hash")
+                    .ok_or("No hash for block")?
+                    .as_str()
+                    .ok_or("Block hash was not string")?,
+            )?;
+            let hash: Hash256 = if hash.len() == 32 {
+                Hash256::from_slice(&hash)
+            } else {
+                return Err(format!("Block has was not 32 bytes: {:?}", hash));
```

Comment:
```suggestion
                return Err(format!("Block hash was not 32 bytes: {:?}", hash));
```

This was existing in the previous code, probably done by me originally!